src/mbgl/renderer/layers/render_symbol_layer.cpp

```diff
@@ -33,6 +33,17 @@ std::unique_ptr<SymbolLayout> RenderSymbolLayer::createLayout(const BucketParame
                                           glyphDependencies);
 }
 
+bool RenderSymbolLayer::updateImpl(Immutable<style::Layer::Impl> baseImpl_) {
+    auto impl_ = dynamicImmutableCast<style::SymbolLayer::Impl>(baseImpl_);
```

Comment:
Instead of introducing yet another cast, can we do `auto& impl_ = dynamic_cast<const style::SymbolLayer::Impl&>(*baseImpl_);`, and just call `std::swap(baseImpl_, baseImpl);` below?